Fix "The user signs up with a profile image" squish test case

alexjba opened the pull request:
### What does the PR do
Fixes #8326

Changes:
1. Update def imagePresent(...) from SquishDriverVerification.py to support image verification in a specific part of the app
See https://doc-snapshots.qt.io/squish/squish-api.html#test-imagepresent-function

2. Update def profile_settings_image_is_updated(self): This verification step was successful even if the profile image was broken in the profile settings view. This is because the profile image was still present in the bottom part of the nav bar. Here we need a more specific image search. For this I have used a search image in the profile settings view only.
The image verification threshold is also lowered to 95 because in the profile settings view the image has an edit button overlapping the image.

3. Update def profile_image_is_updated(self): There are two main updates here. First the threshold is lowered to 95 because  the image can have a user status indicator on top. And second the image verification is done on specific parts of the app (profileNavBarButton and profilePopupImage). Image verification on entire app level could lead to false positive because the image can be shown in different parts of the app at the same time (nav bar and profile popup).
 
4. Update When("the user logs in with password \"|any|\"") step. I had random failures here when the login duration was more than 5 sec (the default of wait_and_get_obj). What was happening is that this step was performed and considered verified after user types password and hit login. The next test case was executed and the test needs an object from AppMain. If the login takes more than 5 sec the test would fail because the object could not be fount within the specified waiting time.

<!-- Fill in the relevant information below to help us evaluate your proposed changes. -->

### Affected areas
SquishDriverVerification -> imagePresent
tst_statusSignUp tests

### Screenshot of functionality (including design for comparison)

https://user-images.githubusercontent.com/47811206/204511904-df9a6af5-4157-42d8-b903-b87af26a7d1e.mov

- [ ] I've checked the design and this PR matches it

<!-- screenshot (or gif/video) that demonstrates the functionality, specially important if it's a bug fix. -->

status-im-auto commented:
### Jenkins Builds
| :grey_question: | Commit | :hash: | Finished (UTC) | Duration | Platform | Result |
|-|-|-|-|-|-|-|
| :heavy_check_mark: | e906e79c | [#1](https://ci.status.im/job/status-desktop/job/branches/job/imports/job/PR-8491/1/) | 2022-11-29 11:08:02 | ~4 min | `imports` | [:page_facing_up:`log`](https://ci.status.im/job/status-desktop/job/branches/job/imports/job/PR-8491/1//consoleText) |

noeliaSD commented on test/ui-test/src/drivers/SquishDriverVerification.py:
```diff
@@ -117,8 +117,8 @@ def verify_or_create_screenshot(vp: str, obj: Dict[str, Any]):
     
-def imagePresent(imageName: str, tolerant: bool = True, threshold: int = 99.5, minScale: int = 50, maxScale: int =  200, multiscale: bool = True):
-    test.imagePresent(imageName, {"tolerant": tolerant, "threshold": threshold, "minScale": minScale, "maxScale": maxScale, "multiscale": multiscale})
+def imagePresent(imageName: str, tolerant: bool = True, threshold: int = 99.5, minScale: int = 50, maxScale: int =  200, multiscale: bool = True, searchRegion: Any = "all"):
```
I know it was already defined like this but could you refactor this method name to follow `lower_case_with_underscores`? We are trying to unify it.

alexjba replied:
Good point. Done

noeliaSD commented on test/ui-test/src/screens/StatusWelcomeScreen.py:
```diff
@@ -222,24 +222,29 @@ def profile_settings_image_is_updated(self):
         type(LoginView.PASSWORD_INPUT.value, password)
         click_obj_by_name(LoginView.SUBMIT_BTN.value)   
+    
+    def waitForMainWindowAfterLogin(self):
```
We already have this validation in a different step so I suggest to remove this method and instead add the following step in the corresponding scenario: 
`When the user logs in with password "TesTEr16843/!@00"`
--> Add this: `And the user lands on the signed in app` --> It makes the validation after the login

alexjba replied:
Cool, didn't notice this step in the test files. Done

noeliaSD commented on test/ui-test/testSuites/global_shared/scripts/global_names.py:
```diff
@@ -1,6 +1,7 @@
 
 statusDesktop_mainWindow = {"name": "mainWindow", "type": "StatusWindow", "visible": True}
 statusDesktop_mainWindow_overlay = {"container": statusDesktop_mainWindow, "type": "Overlay", "unnamed": 1, "visible": True}
+mainWindow_appMain = {"container": statusDesktop_mainWindow, "objectName": "appMain", "type": "Item", "visible": True}
```
Not necessary if we remove the `waitForMainWindowAfterLogin` method!

noeliaSD commented on test/ui-test/testSuites/suite_onboarding/shared/steps/signUpSteps.py:
```diff
@@ -63,6 +63,7 @@ def step(context, seed_phrase):
 @When("the user logs in with password \"|any|\"")
 def step(context, password: str):
     _welcomeScreen.enter_password(password)
+    _welcomeScreen.waitForMainWindowAfterLogin()
```
Not necessary if we add the correspoinding step in the feature file.

noeliaSD commented on ui/main.qml:
```diff
@@ -282,6 +282,7 @@ StatusWindow {
     Component {
         id: app
         AppMain {
+            objectName: "appMain"
```
Not necessary if we remove `waitForMainWindowAfterLogin` method.

noeliaSD commented on test/ui-test/src/screens/StatusWelcomeScreen.py:
```diff
@@ -222,24 +222,29 @@ def profile_settings_image_is_updated(self):
         # first time clicking on settings button closes the my profile modal
```
Unrelated to your changes but it seems this method should be in `StatusMainScreen` instead. I would appreciate if you update it accordingly! :)

alexjba replied:
Moved all the profile verifications to StatusMainScreen

noeliaSD commented on test/ui-test/src/screens/StatusWelcomeScreen.py:
```diff
@@ -222,24 +222,29 @@ def profile_settings_image_is_updated(self):
-        profileNavBarButton = wait_and_get_obj(MainScreenComponents.PROFILE_NAVBAR_BUTTON.value)
-        imagePresent("loginUserName", True, 98, 85, 100, True)
-
+        # open profile popup and check image on profileNavBarButton and profileNavBarPopup
```
Same here. Unrelated to your changes but it seems this method should be in StatusMainScreen instead.

noeliaSD reviewed (changes requested):
Thanks for your first and great contribution in this project! :)

I've added some comments, some not related to your changes but it would be great if you take profit of this PR to update them, as well!

alexjba commented:
> Thanks for your first and great contribution in this project! :)
> 
> I've added some comments, some not related to your changes but it would be great if you take profit of this PR to update them, as well!

Thanks for the review. I've updated the commit. Please take a look
